### PR TITLE
fix(azure)!: Change `azure_compute_skus` PK to calc hash only on `locations` & `capabilities`

### PR DESCRIPTION
**Pull request by candiduslynx**

### CODE REVIEW

File: plugins/source/azure/resources/services/compute/skus.go [hunk 1]
```diff
@@ -19,21 +19,27 @@ func SKUs() *schema.Table {
...
+				Name:            "locations_hash",
 				Type:            schema.TypeString,
-				Resolver:        calcEntryHash,
+				Resolver:        calcLocationsHash,
+				CreationOptions: schema.ColumnCreationOptions{PrimaryKey: true},
+			},
+			schema.Column{
+				Name:            "capabilities_hash",
+				Type:            schema.TypeString,
+				Resolver:        calcCapabilitiesHash,
```
Comment (bbernays) [inline plugins/source/azure/resources/services/compute/skus.go:35]: Is there documentation that justifies this change? And if this is about performance can we benchmark the memory savings and or CPU savings?

Comment (candiduslynx) [reply plugins/source/azure/resources/services/compute/skus.go:35]: No, I just had to inspect the `az vm list-skus` output.
Basically, there are a lot of duplicates around locations and different capabilities. However, this `jq` transformation lets us see the same amount of unique lines:
```bash
az vm list-skus \
  jq -c '.[] | {name, resourceType, locations, capabilities} | (.locations |= sort) | if .capabilities then . else . + {capabilities:[]} end | (.capabilities? |= sort)' \
  sort | wc -l
```

So, this is totally empirical change.

Comment (bbernays) [reply plugins/source/azure/resources/services/compute/skus.go:35]: I don't think we should be introducing breaking changes unless there is a specific benefit for users, and unless we can benchmark the memory improvements I don't think this change rises to that level

Comment (erezrokah) [reply plugins/source/azure/resources/services/compute/skus.go:35]: The work to investigate memory issues on Azure is still on going, see https://github.com/cloudquery/cloudquery/issues/9269.

So far removing `hashstructure` did not show any memory or performance improvements in my internal tests, so I'd put this PR hold for now.

We might want to remove `hashstructure` for other reasons to do it (1. and 3. described in https://github.com/cloudquery/cloudquery/issues/9391#issue-1637475685), but it shouldn't be tied to a specific table.

**Pull request closed without merging**